go-controller/pkg/libovsdbops/template_var.go

```diff
@@ -0,0 +1,110 @@
+	ctx, cancel := context.WithTimeout(context.Background(), types.OVSDBTimeout)
+	defer cancel()
+
+	templatesList := []*nbdb.ChassisTemplateVar{}
+	err := nbClient.List(ctx, &templatesList)
+	if err != nil {
+		return nil, err
+	}
+	return templatesList, err
+}
+
+// CreateOrUpdateChassisTemplateVarOps creates or updates the provided
+// 'template' variable and returns the corresponding ops.
+func CreateOrUpdateChassisTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, template *nbdb.ChassisTemplateVar) ([]libovsdb.Operation, error) {
+
+	modelClient := newModelClient(nbClient)
+	return modelClient.CreateOrUpdateOps(ops, operationModel{
+		Model: template,
+		ModelPredicate: func(item *nbdb.ChassisTemplateVar) bool {
+			return item.Chassis == template.Chassis
+		},
+		OnModelMutations: []interface{}{&template.Variables},
+		ErrNotFound:      false,
+		BulkOp:           false,
+	})
+}
+
+// deleteChassisTemplateVarVariablesOps removes the variables listed as
+// keys of 'template.Variables' and returns the corresponding ops.
+// It applies the mutation to all records that are selected by 'predicate'.
+func deleteChassisTemplateVarVariablesOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, template *nbdb.ChassisTemplateVar,
+	predicate chassisTemplateVarPredicate) ([]libovsdb.Operation, error) {
+
+	deleteTemplate := &nbdb.ChassisTemplateVar{
+		Chassis:   template.Chassis,
+		Variables: map[string]string{},
+	}
+	for name := range template.Variables {
+		deleteTemplate.Variables[name] = ""
+	}
+	modelClient := newModelClient(nbClient)
+	return modelClient.DeleteOps(ops, operationModel{
+		Model:            deleteTemplate,
+		ModelPredicate:   predicate,
+		OnModelMutations: []interface{}{&deleteTemplate.Variables},
+		ErrNotFound:      false,
+		BulkOp:           false,
+	})
+}
+
+// DeleteChassisTemplateVarVariablesOps removes all variables listed as
+// keys of 'template.Variables' from the record matching the same chassis
+// as 'template'. It returns the corresponding ops.
+func DeleteChassisTemplateVarVariablesOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, template *nbdb.ChassisTemplateVar) ([]libovsdb.Operation, error) {
+
+	return deleteChassisTemplateVarVariablesOps(nbClient, ops, template,
+		func(item *nbdb.ChassisTemplateVar) bool {
+			return item.Chassis == template.Chassis
+		})
+}
+
+// DeleteChassisTemplateVarVariables removes the variables listed as
+// keys of 'template.Variables' and commits the transaction to the
+// database. It applies the mutation to all records that are selected by
+// 'predicate'.
+func DeleteChassisTemplateVarVariables(nbClient libovsdbclient.Client, template *nbdb.ChassisTemplateVar, predicate chassisTemplateVarPredicate) error {
+	ops, err := deleteChassisTemplateVarVariablesOps(nbClient, nil, template, predicate)
+	if err != nil {
+		return err
+	}
+
+	_, err = TransactAndCheck(nbClient, ops)
+	return err
+}
```

Comment:
Hmmm this method still looks somewhat confusing to me when compared to the other Delete method. I mean, the doc is clear but ...

What would you think if we named this method `DeleteAllChassisTemplateVarVariables` and instead of passing the predicate, define it inside.

Reply:
Another way to think about it is that this is already kinda special because it just removes variables.

There is other variations I thought about, like passing just list of variable names to remove, Go for it if you think it is better.

Reply:
OK, I updated it.